autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -272,18 +260,9 @@ def _get_train_loss_name(self):
         elif self.problem_type == REGRESSION:
             train_loss_name = 'l2'
         else:
-            raise ValueError("unknown problem_type for LGBModel: %s" % self.problem_type)
+            raise ValueError(f"unknown problem_type for LGBModel: {self.problem_type}")
         return train_loss_name
 
-    def _set_default_searchspace(self):
```

Comment:
LightGBM needs this function because it passes num_classes unlike AbstractModel. Also, it needs to call LightGBM's get_default_searchspace, not AbstractModel's.

Reply:
Perhaps we add _get_default_searchspace function which is called by _set_default_searchspace, so each model only needs to add a 1-liner _get_default_searchspace.

```
def _get_default_searchspace(self):
    return get_default_searchspace(problem_type=self.problem_type, num_classes=self.num_classes)
```

Reply:
> LightGBM needs this function because it passes num_classes unlike AbstractModel. Also, it needs to call LightGBM's get_default_searchspace, not AbstractModel's.

my mistake, I forgot that it is calling LGB's `get_default_searchspace`. The last suggestion is fine, will add the method.